ui-tests/src/test/java/org/openmrs/reference/page/EditReportPage.java

```diff
@@ -0,0 +1,42 @@
+package org.openmrs.reference.page;
+
+import org.openmrs.uitestframework.page.Page;
+import org.openqa.selenium.By;
+
+public class EditReportPage extends Page {
+	
+	private static final By PREVIEW_BUTTON = By.cssSelector("#previewButton");
+	private static final By START_DATE_FIELD = By.cssSelector("#startDate");
+	private static final By END_DATE_FIELD = By.cssSelector("#endDate");
+	private static final By RUN_BUTTON = By.cssSelector("#preview-parameterizable-button");
+	private static final By CLOSE_TAB = By.xpath("//span[contains(text(),'close')]");
+	
+	public EditReportPage(Page parent) {
+	    super(parent);
+	}
+	
+	public void clickOnPreviewButton() {
+	    clickOn(PREVIEW_BUTTON);
+	}
+	
+	public void enterStartDate(String startDate) {
+	    setTextToFieldNoEnter(START_DATE_FIELD, startDate);
+	}
+	
+	public void enterEndDate(String endDate) {
+	    setTextToFieldNoEnter(END_DATE_FIELD, endDate);
+	}
+	
+	public void clickOnRun() {
+	    clickOn(RUN_BUTTON);
+	}
+	
+	public void clickOnCloseTab() {
+	    clickOn(CLOSE_TAB);
+	}
+	
+	@Override
+	public String getPageUrl() {
+	    return "/reporting/reports/reportEditor.form?uuid=b39c4c4c-4881-11e7-a919-92ebcb67fe33";
+	}
```

Comment:
> ohh the uuid is generate at run time. Have cross checked whether uui remained the same

@sherif10 why not getting a better form url than relying on a generated run time which by default can not be guessed on any random build plan?

Reply:
@sherrif10 does it mean the report you gave [above](https://github.com/openmrs/openmrs-distro-referenceapplication/pull/592#discussion_r729654214) was not valid? Or I did not just get the report right?